src/Setup_geometry.jl

```diff
@@ -37,6 +37,24 @@ function flatten_index_dimensions(Phase, ind_vec::Vector{CartesianIndex{3}})
     return ind2D
 end
 
+"""
+    ind2D = flatten_index_dimensions(Phase, ind_vec::Vector{CartesianIndex{3}})
+
+This converts the indices to purely 2D indices if the array `phase` is 2D
+"""
+function flatten_index_dimensions(Phase, ind_vec::Array{Bool, 3})
+    if length(size(Phase))==2
```

Comment:
```suggestion
function flatten_index_dimensions(Phase::AbstractArray{T, N}, ind_vec::Array{Bool, 3}) where {T,N}
    if N==2
```
If `Phase` is an `Array`, this is a bit more elegant and allows the `if` statement to be compiled away.